### PR TITLE
Add Inference API /embed endpoint to July RC branch

### PR DESCRIPTION
**Pull request by aulorbe**
## Problem

Add `/embed` endpoint to July RC branch.

**Note:** 
- This endpoint is only available on the July RC branch (`2024-07-rc`). In order to use the new `/embed` endpoint, users will need to clone this repository, fetch the RC branch, build the project, and play around with the endpoint that way.
- As highlighted in the README, the `/embed` endpoint code only works with Node `>=18`. **Failed integration tests for Node `<18` are expected in this PR**, excluding the run with the `edge` runtime (this runtime provides a `global.fetch` that overrides the ponyfill, so the root issue with Node versions `<18` and how they implement `fetch` is not a problem).

**Included in this PR are:**
- New generated core files
- A new `inference` directory, which contains the code users should use when interacting with this endpoint
- A new `models` directory that contains wrappers around generated data structures. This mimics the organization we have currently in [the Python client](https://github.com/pinecone-io/pinecone-python-client/tree/main/pinecone/models).
- Integration and unit tests

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Infrastructure change (CI configs, etc)
- [x] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

### CODE REVIEW

**Review by gdj0nes — APPROVED** [review]

File: package.json
Comment (gdj0nes) [inline package.json:7]: is this the right link?

Comment (aulorbe) [reply package.json:7]: No, ty!

Comment (aulorbe) [reply package.json]: Fixed!

File: README.md
Comment (aulorbe) [inline README.md:182]: These types of changes come automatically with `npm run format`

Comment (ssmith-pc) [reply README.md:182]: This is a case where I'd suggest holding off on the formatting. It can seem innocuous, but given that these changes will live on the release branch for some period of time and then later merge to main, the formatting updates can make the merge to main difficult if any of the underlying code changes on main between now and then. 

Solvable with merge conflict resolution, but still more trouble than its worth IMO.

We can always do a dedicated formatting PR off of the main branch that just runs the formatter and merge that to main.

File: src/models/embeddingsList.ts
Comment (ssmith-pc) [inline src/models/embeddingsList.ts:135]: Can we rename these function to something like `truncateXXXForDisplay`? Otherwise I think it confuse someone about their intent

File: src/models/embeddingsList.ts
Comment (ssmith-pc) [inline src/models/embeddingsList.ts:60]: Just so I understand how this works, if someone was to want the "original" toString output, is there a way for them to do that?

**Review by ssmith-pc — APPROVED** [review]
This looks great! I have a couple of questions/comments (sorry, I should have asked on the previous review cycle but taking a look with fresh eyes now after the cross-fetch debacle :) )

File: README.md
Comment (ssmith-pc) [reply README.md:182]: I'm also fine leaving the formatting changes in place with this PR -- your call

File: src/models/embeddingsList.ts
Comment (aulorbe) [reply src/models/embeddingsList.ts:60]: I don't believe so, no.

File: README.md
Comment (aulorbe) [reply README.md:182]: I'm pretty sure CI fails w/o these changes (I just run the CI commands themselves locally before pushing, but lemme remove these and see what happens)

Comment (aulorbe) [reply README.md:182]: Yeah, it fails, I have to include these lints: https://github.com/pinecone-io/pinecone-ts-client/actions/runs/9490755109/job/26154849071?pr=230. Reverting.

File: src/models/embeddingsList.ts
Comment (aulorbe) [reply src/models/embeddingsList.ts:135]: Sure!

File: src/models/embeddingsList.ts
Comment (ssmith-pc) [reply src/models/embeddingsList.ts:60]: As long as someone can get to the original/underlying type that is good enough

File: README.md
Comment (ssmith-pc) [reply README.md:182]: sounds good

File: src/models/embeddingsList.ts
Comment (aulorbe) [reply src/models/embeddingsList.ts:60]: Cool :)

**Review by austin-denoble — APPROVED** [review]
🚢

**Pull request merged**
